forgotten `-Wl,` in front of `-headerpad_max_install_names` ?

Have you forgotten `-Wl,` in front of `-headerpad_max_install_names` ? Because we have now https://github.com/sagemath/sage/issues/40106 ...

_Originally posted by @dimpase in https://github.com/gap-system/gap/pull/5769#discussion_r2090093581_
            
As well, adding `-Wl,-headerpad_max_install_names` in `build/gap-install` linker call appears to be necessary.

A fix proposed on #5992

It could be the case that for reproducing one needs arm64 Darwin 24.4.0. I can't reproduce the Sage issue on 24.3.0.

Actually, fixing this  doesn't fix the Sage issue - for the latter we apparently have to directly add  `-Wl,-headerpad_max_install_names` to LDFLAGS before calling `make install`, specifically, when `build/gap-install` is invoked by `install-bin`. Otherwise, it fails as follows:

```
gcc         -pthread -lpthread -L/Users/palmieri/Sage/TESTING/sage-10.7.beta3/local/lib -L/Users/palmieri/Sage/TESTING/sage-10.7.beta3/local/lib  build/obj/build/main.c.o  -lgmp -lz -lreadline     -L/Users/palmieri/Sage/TESTING/sage-10.7.beta3/local/var/tmp/sage/build/gap-4.14.0/src -lgap -o build/gap-install
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: changing install names or rpaths can't be redone for: build/gap-install (for architecture arm64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
make[3]: *** [build/gap-install] Error 1
```
with LDFLAGS set, this call looks like
```
gcc         -pthread -lpthread -Wl,-headerpad_max_install_names -L/Users/palmieri/Sage/TESTING/sage-10.7.beta3/local/lib -L/Users/palmieri/Sage/TESTING/sage-10.7.beta3/local/lib  build/obj/build/main.c.o  -lgmp -lz -lreadline     -L/Users/palmieri/Sage/TESTING/sage-10.7.beta3/local/var/tmp/sage/build/gap-4.14.0/src -lgap -o build/gap-install
```

The only difference is that `-Wl,-headerpad_max_install_names` is added as the 3rd argument to the call to gcc (in link mode. 
This is not a surprise, as `$(LINK_SHLIB_FLAGS)` only appears in the libgap-making target, but not in `build/gap-install`.

Thus, there is another bug, not only missing ` -Wl,`, but also missing `-Wl,-headerpad_max_install_names` when `build/install` is invoked in `install-bin`.

--- a/Makefile.rules
+++ b/Makefile.rules
@@ -431,7 +431,7 @@ build/obj/%.c.o: %.c cnf/GAP-CFLAGS cnf/GAP-CPPFLAGS $(obj_deps)
 ########################################################################
 
 LINK=$(CC)
-
+GAP_INSTALL_EXTRAFLAGS =
 SHLIB_MAJOR = $(GAP_KERNEL_MAJOR_VERSION)
 ifneq (,$(findstring cygwin,$(host_os)))
   SHLIB_EXT=.dll
@@ -447,7 +447,8 @@ else ifneq (,$(findstring darwin,$(host_os)))
   LINK_SHLIB_FLAGS += -compatibility_version $(LIBGAP_COMPAT_VER)
   LINK_SHLIB_FLAGS += -current_version $(LIBGAP_CURRENT_VER)
   LINK_SHLIB_FLAGS += -Wl,-single_module
-  LINK_SHLIB_FLAGS += -headerpad_max_install_names
+  LINK_SHLIB_FLAGS += -Wl,-headerpad_max_install_names
+  GAP_INSTALL_EXTRAFLAGS = -Wl,-headerpad_max_install_names
 
   GAP_CPPFLAGS += -DPIC
   GAP_CFLAGS += -fno-common
@@ -519,7 +520,7 @@ build/main.c: src/main.c config.status
 
 # build rule for the gap executable used by the `install-bin` target
 build/gap-install: build/obj/build/main.c.o libgap$(SHLIB_EXT) cnf/GAP-LDFLAGS cnf/GAP-LIBS cnf/GAP-OBJS
-	$(QUIET_LINK)$(LINK) $(GAP_LDFLAGS) $< $(GAP_LIBS) -L${abs_builddir} -lgap -o $@
+	$(QUIET_LINK)$(LINK) $(GAP_LDFLAGS) $(GAP_INSTALL_EXTRAFLAGS) $< $(GAP_LIBS) -L${abs_builddir} -lgap -o $@
 	$(INSTALL_NAME_TOOL) -change $(LIBGAP_FULL) $(libdir)/$(LIBGAP_FULL) $@
 
 endif

